src/vos/tests/vts_mvcc.c

```diff
@@ -164,8 +164,7 @@ set_oid(int i, char *path, daos_unit_oid_t *oid)
 	 *   - Lower 8 bits are set to the numeric value of path[L_O].
 	 */
 	oid->id_pub.hi = 0;
-	D_ASSERT(L_O < strlen(path));
-	oid->id_pub.lo = (i << 8) + path[L_O];
+	oid->id_pub.lo = (i << 8) + 'o';
 	daos_obj_set_oid(&oid->id_pub, DAOS_OT_MULTI_UINT64, OR_RP_1, 1, 0);
```

Comment:
Could we replace the change to this file (vts_mvcc.c) with the following?

```
diff --git a/src/vos/tests/vts_mvcc.c b/src/vos/tests/vts_mvcc.c
index 41e9b1b0d..e1a54712f 100644
--- a/src/vos/tests/vts_mvcc.c
+++ b/src/vos/tests/vts_mvcc.c
@@ -1213,6 +1213,7 @@ conflicting_rw_exec_one(struct io_test_args *arg, int i, int j, bool empty,
        if (!empty) {
                char    pp[L_COUNT + 1] = "coda";
 
+               D_ASSERT(strlen(rp) <= L_COUNT);
                memcpy(pp, rp, strlen(rp));
                print_message("  update(%s, "DF_X64") before %s(%s, "
                              DF_X64"): ", pp, re - 1, r->o_name, rp, re);
@@ -1475,8 +1476,8 @@ uncertainty_check_exec_one(struct io_test_args *arg, int i, int j, bool empty,
                char            pp[L_COUNT + 1] = "coda";
                daos_epoch_t    pe = ae - 1;
 
-               D_ASSERT(strnlen(wp, L_COUNT) <= sizeof(pp) - 1);
-               memcpy(pp, wp, strnlen(wp, L_COUNT));
+               D_ASSERT(strlen(rp) <= L_COUNT);
+               memcpy(pp, wp, strlen(wp));
                print_message("  update(%s, "DF_U64") (expect DER_SUCCESS): ",
                              pp, pe);
                rc = update_f(arg, NULL /* txh */, pp, pe);
```

I think the previous strnlen changes were incorrect. And this time the issue looks like a false positive.

Reply:
What I don't understand here is why pp[L_COUNT + 1] is set to "coda" ?

What is the point of the memcpy of rp to pp.  rp is always some substring of "coda" and we are not copying the NULL character so the resulting string will always be "pp", no?

Reply:
The string rp points to may not be prefix of "coda" if we change the tests/subtests (at least that was the idea). So memcpy here means: Construct a path to the akey level that use rp as a prefix. For instance, if rp == "cO", then we fill out the akey level and produce "cOda".